src/shared/version_op.c

```diff
@@ -661,11 +661,27 @@ os_info *get_unix_version()
             } else if(strcmp(strtok_r(buff, "\n", &save_ptr),"Darwin") == 0){
                 info->os_platform = strdup("darwin");
 
-                if (cmd_output_ver = popen("sw_vers -productName", "r"), cmd_output_ver) {
-                    if(fgets(buff, sizeof(buff) - 1, cmd_output_ver) == NULL){
-                        mdebug1("Cannot read from command output (sw_vers -productName).");
-                    } else {
-                        w_strdup(strtok_r(buff, "\n", &save_ptr), info->os_name);
+                if (cmd_output_ver = popen("system_profiler SPSoftwareDataType", "r"), cmd_output_ver) {
+                    while (fgets(buff, sizeof(buff) - 1, cmd_output_ver)) {
+                        char *key = strtok_r(buff, ":", &save_ptr);
+                        if (key) {
+                            char *trimed_key = os_strip_char(key, ' ');
+                            if (strcmp(trimed_key, "SystemVersion") == 0) {
```

Comment:
Check if trimmed_key is not NULL
Try to use safe functions like strncmp https://github.com/wazuh/wazuh/wiki/Coding-style-guide-%28C%29#secure-coding